libraries/protocol/include/golos/protocol/worker_proposal_operations.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <golos/protocol/authority.hpp>
+#include <golos/protocol/base.hpp>
+
+namespace golos { namespace protocol {
+
+    enum worker_proposal_type {
+        task,
+        premade_work,
+        _wpt_size
+    };
+
+    struct worker_proposal_operation : public base_operation {
+        account_name_type author;
+        std::string permlink;
+        worker_proposal_type type = task;
+
+        extensions_type extensions;
+
+        void validate() const;
+
+        void get_required_active_authorities(flat_set<account_name_type>& a) const {
```

Comment:
why `active`?
it can be active if requires some fee or so. actually, making this `active` prevents combining it to the same transaction with post creation